entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityOperationService.kt

```diff
@@ -58,6 +57,87 @@ class EntityOperationService(
+     * linked to invalid entity).
+     */
+    fun update(entities: List<ExpandedEntity>, createBatchResult: BatchOperationResult): BatchOperationResult {
+        val existingEntitiesIds = createBatchResult.success.plus(entities.map { it.id })
+        val nonExistingEntitiesIds = createBatchResult.errors.map { it.entityId }
+        return entities.parallelStream().map { entity ->
+            updateEntity(entity, existingEntitiesIds, nonExistingEntitiesIds)
+        }.collect(
+            { BatchOperationResult() },
+            { batchOperationResult, updateResult ->
+                updateResult.fold({
+                    batchOperationResult.errors.add(it)
+                }, {
+                    batchOperationResult.success.add(it)
+                })
+            },
+            BatchOperationResult::plusAssign
+        )
+    }
+
+    private fun updateEntity(
+        entity: ExpandedEntity,
+        existingEntitiesIds: List<String>,
+        nonExistingEntitiesIds: List<String>
+    ): Either<BatchEntityError, String> {
+        // All new attributes linked entities should be existing in the DB.
+        val linkedEntitiesIds = entity.getLinkedEntitiesIds()
+        val invalidLinkedEntityId =
+            findInvalidEntityId(linkedEntitiesIds, existingEntitiesIds, nonExistingEntitiesIds)
+
+        // If there's a link to an invalid entity, then avoid calling the processor and return an error
+        if (invalidLinkedEntityId != null) {
+            return Either.left(
+                BatchEntityError(
+                    entity.id,
+                    arrayListOf("Target entity $invalidLinkedEntityId does not exist.")
+                )
+            )
+        }
+
+        return try {
+            val (_, notUpdated) = entityService.appendEntityAttributes(
+                entity.id,
+                entity.attributes,
+                false
+            )
+
+            if (notUpdated.isEmpty()) {
+                Either.right(entity.id)
+            } else {
+                Either.left(
+                    BatchEntityError(
+                        entity.id,
+                        ArrayList(notUpdated.map { it.attributeName + " : " + it.reason })
+                    )
+                )
+            }
+        } catch (e: BadRequestDataException) {
+            Either.left(BatchEntityError(entity.id, arrayListOf(e.message)))
+        }
+    }
+
+    private fun findInvalidEntityId(
+        entitiesIds: List<String>,
+        existingEntitiesIds: List<String>,
+        nonExistingEntitiesIds: List<String>
+    ): String? {
+        val invalidEntityId = entitiesIds.intersect(nonExistingEntitiesIds).firstOrNull()
+        if (invalidEntityId == null) {
+            val unknownEntitiesIds = entitiesIds.minus(existingEntitiesIds)
+            return unknownEntitiesIds
+                .minus(neo4jRepository.filterExistingEntitiesIds(unknownEntitiesIds)).firstOrNull()
```

Comment:
if `unkownEntitiesIds` is empty, you can probably avoid the query in the repository

Reply:
Maybe this is a bad approach but in the Neo4JRepository method I handled this check to avoid a call to DB.
I did it this way else you need to check everytime you call the filterExistingEntitiesIds method

Reply:
OK so.